Introduce `Sensitive` which can be used for hiding `Debug` impls

hrxi opened the pull request:
Use it everywhere where we used to manually implement `Debug` on types to hide information.

Alternative to #2463.

Fixes #2459.

sisou commented on lib/src/config/config_file/mod.rs:
```diff
@@ -252,7 +253,7 @@ pub struct MetricsServerSettings {
     pub bind: Option<String>,
     pub port: Option<u16>,
     pub username: Option<String>,
```
If RPC username is sensitive, then metrics server username should, too?

hrxi replied:
Done, but unsure if usernames should be hidden or not, see below.

hrxi commented:
Yea, exactly what @sisou says. I'm interested in whether we want to hide usernames or not. Previously, we'd hide them.

hrxi committed:
Introduce `Sensitive` which can be used for hiding `Debug` impls

Use it everywhere where we used to manually implement `Debug` on types
to hide information.

Alternative to #2463.

Fixes #2459.

hrxi commented:
After a quick discussion, we're showing all usernames now.

jsdanielh reviewed (approved):
LGTM
